memecity.game/BSPGenerator.h

```diff
@@ -0,0 +1,49 @@
+#ifndef _BSP_GENERATOR_H
+#define _BSP_GENERATOR_H
+#include <vector>
+#include <iostream>
+#include <time.h>
+#include "GeneratorStrategy.h"
+
+struct Node : Rectangle {
+	Node *left, *right;
+	Node(int x, int y, int w, int h) : Rectangle(x, y, w, h)
+	{
+		left = nullptr;
+		right = nullptr;
+	}
+
+	bool has_left()
+	{
+		return left != NULL;
+	}
+	bool has_right()
+	{
+		return right != NULL;
```

Comment:
This NULL check is deprecated. It will be a better practice to check if it's not a nullptr. This is applies for the `bool has_left()` as well.
```suggestion
		return right != nullptr;
```